pkg/consensus/consensus.go

```diff
@@ -334,6 +360,33 @@ events:
 	close(s.finished)
 }
 
+func (s *service) resetWatchdogTimer(threshold time.Duration) {
+	lastTimestamp := int64(s.lastTimestamp)
+	now := time.Now().UnixMilli()
+	delta := time.Millisecond * time.Duration(lastTimestamp-now)
+	resetAfter := delta
+	for {
+		resetAfter += threshold
+		if resetAfter > 0 {
+			break
+		}
+	}
+	if s.watchdogTimer == nil {
+		// Don't reinitialize consensus for the first time, the service has just started.
+		s.watchdogTimer = time.NewTimer(resetAfter)
+	} else {
+		s.watchdogTimer.Reset(resetAfter)
+		if delta+threshold <= 0 {
+			s.log.Warn("couldn't accept new block, sending signal to restart consensus service",
```

Comment:
To simplify grep, could we use something like "DBFTWatchdog triggered after X sec from last height Y.  Sending signal to restart consensus service."

Reply:
I'm not sure it's a good idea to put `X` and `Y` right into the log message. Can we keep them as log fields?

Currently, logs look like
```
2022-06-28T15:18:42.921Z	WARN	couldn't accept new block, sending signal to restart consensus service	{"latest block index": 18, "latest block timestamp": 1656429502920, "time since latest block": "20.001s", "time until next restart": "9.999s"}
```
Isn't it enough for grepping?

Reply:
Yes, looks good. Missed the lines following the log message =)